chain/chain/src/chain.rs

```diff
@@ -1529,6 +1560,18 @@ impl Chain {
         ),
         Error,
     > {
+        // We set a limit to the max number of blocks that we will be processing at the same time.
+        // Since processing a block requires that the its previous block is processed, this limit
+        // is likely never hit, unless there are many forks in the chain.
+        // In this case, we will simply drop the block.
+        if self.processing_blocks.is_full() {
```

Comment:
do we need this here? or should we simply fail later (when trying to add the preprocessing info)?

Reply:
Alternatively, maybe fail *only* here, and make the implementation of `processnig_blocks` infailable? That is, don't put the limit *inside* the pool, instead provide the `.len()` fn and let the caller to enforce the limit?